### PR TITLE
C# Performance improvements

### PR DESCRIPTION
**Pull request by peter-quix**
- Use ArraySegment at more places to avoid unnecessary byte[] initialization and copying
- Implement custom JSON serializer
- Minor improvements for buffer array concat

### CODE REVIEW

**Comment by peter-quix** [comment]
Compound effect is 10-20% performance gain when using JSON without introducing breaking changes to the protocol.

API has breaking changes in Transport, but it is not expected to be used anywhere else

File: src/CsharpClient/QuixStreams.Telemetry/Models/Telemetry/Parameters/Codecs/TimeseriesDataCustomJsonCodec.cs [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        }
+
+        private void Serialize2(StreamWriter streamWriter, TimeseriesDataRaw obj)
+        {
+            using (var jsonWriter = new JsonTextWriter(streamWriter))
+            {
+                jsonWriter.CloseOutput = false;
+                new JsonSerializer().Serialize(jsonWriter, obj);
+            }
+        }
```
Comment (harisbotic) [inline src/CsharpClient/QuixStreams.Telemetry/Models/Telemetry/Parameters/Codecs/TimeseriesDataCustomJsonCodec.cs]: Serialize2 doesn't seem to be used, and Serialize1 could have a better name

Comment (peter-quix) [reply src/CsharpClient/QuixStreams.Telemetry/Models/Telemetry/Parameters/Codecs/TimeseriesDataCustomJsonCodec.cs]: fixed!

**Review by harisbotic — APPROVED** [review]
LGTM

**Pull request merged**